tests/test_query_engine_dsl.py

```diff
@@ -391,3 +393,274 @@ def test_date_arithmetic_convert_to_weeks(engine, cohort_with_population):
+
+    age_in_1990 = reference_date - dob  # -> DateDeltaSeries
+    age_in_2000 = reference_date1 - dob  # -> DateDeltaSeries
+
+    data_definition.age_in_days_1990 = age_in_1990.convert_to_days()
+    data_definition.age_in_days_2000 = age_in_2000.convert_to_days()
+
+    data_definition.age_diff = age_in_2000 - age_in_1990
+
+    result = engine.extract(data_definition)
+    assert result == [
+        {
+            "patient_id": 1,
+            "age_in_days_1990": 31,
+            "age_in_days_2000": 3684,
+            "age_diff": 3684 - 31,
+        },
+        {
+            "patient_id": 2,
+            "age_in_days_1990": 1096,
+            "age_in_days_2000": 4749,
+            "age_diff": 4749 - 1096,
+        },
+    ]
+
+
+def test_date_arithmetic_subtract_datedeltaseries_and_integer(
+    engine, cohort_with_population
+):
+    input_data = [
+        patient(1, dob="1990-08-10"),
+        patient(2, dob="1987-09-10"),
+    ]
+    engine.setup(input_data)
+
+    patients = tables.patients
+    data_definition = cohort_with_population
+    reference_date = "1990-09-10"
+    dob = patients.select_column(patients.date_of_birth)  # -> DateSeries
+
+    age = reference_date - dob  # -> DateDeltaSeries
+    data_definition.age_in_days = age.convert_to_days()
+    data_definition.age_minus_10 = age - 10
+    data_definition.two_thousand_minus_age = 2000 - age
+
+    result = engine.extract(data_definition)
+    assert result == [
+        {
+            "patient_id": 1,
+            "age_in_days": 31,
+            "age_minus_10": 21,
+            "two_thousand_minus_age": 1969,
+        },
+        {
+            "patient_id": 2,
+            "age_in_days": 1096,
+            "age_minus_10": 1086,
+            "two_thousand_minus_age": 904,
+        },
+    ]
+
+
+def test_date_arithmetic_add_and_subtract(engine, cohort_with_population):
+    input_data = [
+        patient(1, dob="1990-08-10"),
+    ]
+    engine.setup(input_data)
+
+    patients = tables.patients
+    data_definition = cohort_with_population
+    reference_date = "1990-09-10"
+    reference_date1 = "1991-09-10"
+    dob = patients.select_column(patients.date_of_birth)  # -> DateSeries
+
+    age_in_1990 = reference_date - dob  # -> DateDeltaSeries
+    age_in_1991 = reference_date1 - dob  # -> DateDeltaSeries
+
+    data_definition.age_in_days_1990 = age_in_1990.convert_to_days()
+    data_definition.age_in_days_1991 = age_in_1991.convert_to_days()
+
+    data_definition.age_diff_plus_10 = age_in_1991 - age_in_1990 + 10
+    data_definition.dob_plus_age_diff_plus_10 = dob + age_in_1991 - age_in_1990 + 10
+
+    result = engine.extract(data_definition)
+    assert result == [
+        {
+            "patient_id": 1,
+            "age_in_days_1990": 31,
+            "age_in_days_1991": 396,
+            "age_diff_plus_10": 375,
+            "dob_plus_age_diff_plus_10": date(1991, 8, 20),
+        }
+    ]
```

Comment:
These tests look sensible, but it's hard as a reader to evaluate whether they're correct, because they're full of magic numbers (and magic dates!).  For instance, where do `396` or `date(1991, 8, 20)` come from?

Something to consider for the future might be to do something like:

```
def test_date_arithmetic_add_and_subtract(engine, cohort_with_population):
    dob = date(1990, 8, 10)
    reference_date = date(1990, 9, 10)

    input_data = [
        patient(1, dob=str(dob),
    ]
    engine.setup(input_data)

    patients = tables.patients
    data_definition = cohort_with_population
    dob_series = patients.select_column(patients.date_of_birth)  # -> DateSeries

    age_in_1990 = str(reference_date) - dob_series  # -> DateDeltaSeries
    data_definition.age_in_days_1990 = age_in_1990.convert_to_days()

    result = engine.extract(data_definition)
    assert result == [
        {
            "patient_id": 1,
            "age_in_days_1990": (reference_date - dob).days,
        }
    ]
```

(I've probably made several mistakes here, but hopefully you get the idea!)

Reply:
good point, it would make sense to use python date subtractions in the tests.  I think I avoided it because of the custom differences we have (e.g. weeks as whole weeks rather than boundaries crossed).  But in the adding/subtracting deltas, we're only using days, so that's not an issue.  I'll take a look at updating the tests.